src/unicodeScripts.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+/**
+ * This file defines the Unicode scripts and script families that we
+ * support.  To add new scripts or families, just map the script or family
+ * name to an array of numbers. The arrays must have an even number of
+ * elements and each element pair must define the start and end (inclusive)
+ * of a block of codepoints that are part of the script.
+ */
+const scriptData = {
+    // Chinese and Japanese.
+    // The "k" in cjk is for Korean, but we've separated Korean out
+    cjk: [
+        0x3000, 0x30FF,  // CJK symbols and punctuation, Hiragana, Katakana
+        0x4E00, 0x9FAF,  // CJK ideograms
+        0xFF00, 0xFF60,  // Fullwidth punctuation
+        // TODO: add halfwidth Katakana and Romanji glyphs
```

Comment:
Maybe change these to:
```
cjk: [
    [0x3000, 0x30FF],
    [0x4E00, 0x9FAF],
    [0xFF00, 0xFF60],
],
brahmic: [ 
    [0x0900, 0x109F],
],
```

Reply:
It simplifies the logic in `scriptFromCodepoint` and makes it more obvious what's going on when there are multiple ranges.

Reply:
I ended up changing this all around to make Flow happy and to speed up perf. (I was making those changes before you started the review... sorry about that.) But now that I've made codepointSupported() fast, I can probably make this change to make the data structure clearer...